### PR TITLE
[frontend] Improving stability of unit tests

### PR DESCRIPTION
**Pull request by siemiatj**
Will have to run it in Jenkins a bunch of times and see if it's any better.

### CODE REVIEW

**Comment by siemiatj** [comment]
Ok I've run it 17 times and there were 3 failures: one because of linter complaining and 2 caused by Jenkins going bananas. Unit tests passed every single time.

File: frontend/src/__tests__/components/table/TableQuickInput.test.js [hunk 1]
```diff
@@ -1,56 +1,52 @@
...
-import { mount } from 'enzyme';
-// import { ShortcutProvider } from '../../../components/keyshortcuts/ShortcutProvider';
-import { initialState as appHandlerState } from '../../../reducers/appHandler';
-import { initialState as windowHandlerState } from '../../../reducers/windowHandler';
-import { Provider } from 'react-redux';
-import configureStore from 'redux-mock-store';
-import merge from 'merge';
-import tableQuickInputProps from '../../../../test_setup/fixtures/table/table_quick_input.json';
+import { shallow, render } from 'enzyme';
+import renderer from 'react-test-renderer';
```
Comment (petrican) [inline frontend/src/__tests__/components/table/TableQuickInput.test.js:3]: I believe L:3 can be removed

File: frontend/src/__tests__/components/table/TableQuickInput.test.js [hunk 2]
```diff
@@ -1,56 +1,52 @@
...
-    expect(html).toContain(
-      `<form class="row quick-input-container"><div class="col-sm-12 col-md-3 col-lg-2 hint">(Press 'Enter' to add)</div><button type="submit" class="hidden-up"></button></form>`
+    nock(config.API_URL)
+      .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
+      .get(`/window/${docType}/${docId}/${tabId}/quickInput/layout`)
+      .reply(200, quickInputData.layout1); 
+  });
+
+  it('renders without errors', () => {
+    const wrapperTableCMenu = shallow(
```
Comment (petrican) [inline frontend/src/__tests__/components/table/TableQuickInput.test.js:35]: const wrapperTableCMenu = shallow(<TableQuickInput {...initialProps} />);  // prettify it

File: frontend/src/__tests__/components/table/TableQuickInput.test.js [hunk 3]
```diff
@@ -1,56 +1,52 @@
...
+    expect(wrapperTableCMenu.find('.quick-input-container').length).toBe(1);
+
+    // const nockCalls = nock.recorder.play() // "play" the recording into a variable
+    // console.log(`Nock Captured Calls: \n${JSON.stringify(nockCalls,null,2)}`) // inspect calls that Nock recorder
+
+    // nock.restore();
   });
 
+  it('renders without error 22s', () => {
+    const wrapperTableCMenu = render(
```
Comment (petrican) [inline frontend/src/__tests__/components/table/TableQuickInput.test.js:47]: const wrapperTableCMenu = render(<TableQuickInput {...initialProps} />);  // prettify it in one line pls also

**Comment by petrican** [comment]
Note: To replicate I ran the entire suite (npm run test) several times and not just that solo test (npm run test:watch)

**Comment by petrican** [comment]
After doing a bit of investigation I strongly believe the issue originates from the createWaitForElement from the enzyme-wait. Btw that lib hasn't been updated for few years by the contributor. I have some doubts  about its reliability

**Comment by petrican** [comment]
I have pushed changes that are fixing the flaky tests. Will merge this after build completes. There are some issues with jenkins and it is under maintenance.

**Pull request merged**